### PR TITLE
Fix session retry error analysis to account for timeout errors

### PR DESCRIPTION
**Pull request by jfcloutier**

### CODE REVIEW

File: lib/jackalope/session.ex [hunk 1]
```diff
@@ -268,11 +268,12 @@ defmodule Jackalope.Session do
 
   defp expired?(expiration), do: Expiration.after?(Expiration.expiration(0), expiration)
 
-  defp retry_error?({{:subscribe, _}, _}, reason) do
-    # reason is a list of {status, {topic, qos}}
-    # if all the statuses are :ok or :access_denied, we don't want to retry
-    Enum.any?(reason, fn {status, _topic} -> status not in [:ok, :access_denied] end)
+  # When reasons is a list of {status, {topic, qos}},
```
Comment (fhunleth) [inline lib/jackalope/session.ex:271]: The error responses here are a little strange for typical Elixir functions. Are these handling the return value of one Tortoise function?

File: lib/jackalope/session.ex [hunk 2]
```diff
@@ -268,11 +268,12 @@ defmodule Jackalope.Session do
...
+  # if all the statuses are :ok or :access_denied, we don't want to retry
+  defp retry_error?({{:subscribe, _}, _}, reasons) when is_list(reasons) do
+    Enum.any?(reasons, fn {status, _topic} -> status not in [:ok, :access_denied] end)
   end
 
+  # Else, we want to retry
```
Comment (fhunleth) [inline lib/jackalope/session.ex:277]: Is this just for `:timeout` errors? or are there others?

File: lib/jackalope/session.ex (hunk 1 above)
Comment (jfcloutier) [reply lib/jackalope/session.ex:271]: Tortoise311 is a little strange. That's how it sends subscription errors to Jackalope's TortoiseClient, as tuple messages. ToirtoiseClient forwards them to Session for reporting. The error component from Tortoise311 is either a list of reasons or a :timeout error. The :timeout case was not being handled.

File: lib/jackalope/session.ex (hunk 2 above)
Comment (jfcloutier) [reply lib/jackalope/session.ex:277]: Not as far as subscriptions are concerned, from looking at Tortoise311.Connection  from where these error messages are sent.

**Review by fhunleth — APPROVED** [review]

**Pull request merged**